sklearn/tests/test_pipeline.py

```diff
@@ -277,14 +277,19 @@ def test_fit_predict_on_pipeline():
     # transform and clustering steps separately
     iris = load_iris()
     scaler = StandardScaler()
+    scaler_for_pipeline = StandardScaler()
```

Comment:
Please add a comment that this is necessary since `Pipeline` does not clone the estimators.